Classes/Controller/ImageRenderingController.php

```diff
@@ -78,16 +79,19 @@ public function renderImageAttributes(?string $content, array $conf = []): strin
                     $resourceFactory = GeneralUtility::makeInstance(ResourceFactory::class);
                     $systemImage     = $resourceFactory->getFileObject($fileUid);
 
-                    if ($imageSource !== $systemImage->getPublicUrl()) {
-                        // Source file is a processed image
+                    // check if there is a processed variant, if not, create one
+                    $processedFile = null;
+                    if ($systemImage instanceof File) {
+                        /** @var \Netresearch\RteCKEditorImage\Utils\ProcessedFilesHandler $checkProcessed */
+                        $checkProcessed = GeneralUtility::makeInstance(\Netresearch\RteCKEditorImage\Utils\ProcessedFilesHandler::class);
                         $imageConfiguration = [
                             'width'  => (int) ($imageAttributes['width']  ?? $systemImage->getProperty('width') ?? 0),
                             'height' => (int) ($imageAttributes['height'] ?? $systemImage->getProperty('height') ?? 0),
                         ];
+                        $processedFile = $checkProcessed->createProcessedFile($systemImage, $imageConfiguration);
+                    }
 
-                        $processedFile = $this->getMagicImageService()
-                            ->createMagicImage($systemImage, $imageConfiguration);
-
+                    if ($processedFile !== false) {
```

Comment:
`$processedFile` may be `null` (see line 83) and this will evaluate to `true` if `$processedFile` is `null`

Reply:
done